### PR TITLE
Fix routing service registration

### PR DESCRIPTION
**Pull request by rynowak**
/cc @pranavkm

### CODE REVIEW

File: src/Microsoft.AspNet.Routing/DependencyInjection/RoutingServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -11,24 +11,27 @@ namespace Microsoft.Extensions.DependencyInjection
...
+            services.TryAddSingleton(UrlEncoder.Default);
+
+            if (configureOptions != null)
             {
-                throw new ArgumentNullException(nameof(setupAction));
+                services.Configure(configureOptions);
             }
 
-            services.TryAddSingleton(UrlEncoder.Default);
-            services.Configure(setupAction);
```
Comment (rynowak) [inline src/Microsoft.AspNet.Routing/DependencyInjection/RoutingServiceCollectionExtensions.cs]: OOPS

Comment (rynowak) [inline src/Microsoft.AspNet.Routing/DependencyInjection/RoutingServiceCollectionExtensions.cs]: This should have been added in `AddRouting`, and there shouldn't be a `ConfigureRouting`

File: src/Microsoft.AspNet.Routing/DependencyInjection/RoutingServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -11,24 +11,27 @@ namespace Microsoft.Extensions.DependencyInjection
...
     {
-        /// <summary>
-        /// Configures a set of <see cref="RouteOptions"/> for the application.
-        /// </summary>
-        /// <param name="services">The services available in the application.</param>
-        /// <param name="setupAction">An action to configure the <see cref="RouteOptions"/>.</param>
-        public static void ConfigureRouting(
+        public static IServiceCollection AddRouting(this IServiceCollection services)
+        {
+            return AddRouting(services, null);
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Routing/DependencyInjection/RoutingServiceCollectionExtensions.cs]: `configureOptions: null`

File: src/Microsoft.AspNet.Routing/DependencyInjection/RoutingServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -11,24 +11,27 @@ namespace Microsoft.Extensions.DependencyInjection
...
+        }
+
+        public static IServiceCollection AddRouting(
             this IServiceCollection services,
-            Action<RouteOptions> setupAction)
+            Action<RouteOptions> configureOptions)
         {
-            if (setupAction == null)
+            services.AddOptions();
+            services.TryAddTransient<IInlineConstraintResolver, DefaultInlineConstraintResolver>();
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Routing/DependencyInjection/RoutingServiceCollectionExtensions.cs]: :+1: Yeah I saw this was missing too but wasn't sure how it got magically added.

**Comment by pranavkm** [comment]
:shipit:

**Pull request closed without merging**